fairscale/experimental/nn/multiprocess_pipe.py

```diff
@@ -97,83 +97,71 @@ def dloss(input_rref: rpc.RRef, target_rref: rpc.RRef) -> rpc.RRef:
 
 
 class MultiProcessPipe(Module):
-    """Wraps an arbitrary :class:`nn.Sequential <torch.nn.Sequential>` module
-    to train on Pipe_. If the module requires lots of memory, Pipe will be
+    """Paritions a sequential list of modules across multiple workers to train using a
+    MultiProcessPipe_. If the module requires lots of memory, MultiProcessPipe will be
     very efficient.
     ::
 
-        model = nn.Sequential(a, b, c, d)
-        model = Pipe(model, balance=[1, 1, 1, 1], chunks=8)
+        devices = ["worker0/cuda:0", "worker1/cuda:1"]
+        # LayerSpec format: [("name", nn.Module, args, kwargs), ...]
+        layer_spec = [("linear", nn.Linear, (4, 4), {}), ("relu", nn.ReLU, (), {})]
```

Comment:
Why can't we pass the model in as before? Any reason for having LayerSpecs?

Reply:
Passing in a nn.Module would require instantiating the Module into CPU memory of the main worker. For a really large model (> 1T param), that will consume more memory than is available. We would also need to transfer parts of the model over the network which would be very expensive. It's much more efficient to instantiate and create sub-graphs on the final worker.